server/aesdsocket.c

```diff
@@ -0,0 +1,216 @@
+#include <stdio.h>
+#include <strings.h>
+#include <unistd.h> 
+#include <sys/socket.h>
+#include <syslog.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <error.h>
+#include <signal.h>
+#include <arpa/inet.h>
+#include <stdlib.h>
+#include <string.h>
+
+void sig_handler(int signal)
+{
+	if(unlink("/var/tmp/aesdsocketdata")==-1)
+		perror("unlink failed");
+		
+	syslog(LOG_INFO,"Caught signal,exiting");
+	exit(0);
+}
+void connection_handler(int* arg)
+{
+
+        int connfd = *arg,bytes;
+        int pkt_length=0;
+        unsigned int rd_bytes;
+        char buff[512],rd_buff[128],*packet=NULL;
+        free(arg);
+
+        int fd,rd_fd;
+
+        fd = open("/var/tmp/aesdsocketdata",O_RDWR|O_CREAT|O_APPEND,S_IRUSR|S_IWUSR);
+	if(fd==-1)
+	{
+		perror("open call failed");
+		exit(-1);
+	}
+	
+	while((bytes=recv(connfd,buff,sizeof(buff),0))!=0)
+	{
+		int written=0;
+		char *sop=buff,*eop=NULL; // store pointer to beginning of buffer		
+		while(written < bytes)	// loop until all bytes are written 
+		{
+			
+			if((eop=strchr(sop,'\n'))>0) // if '\n' found write into file
+			{	
+				if(packet==NULL)			// create packet if not created
+					packet = malloc(eop-sop+1);
+				else					// resize packet if already created
+					packet = realloc(packet,pkt_length+eop-sop+1);
```

Comment:
@lnxblog what assumption is made here about the size of RAM heap relative to the total size of all packets sent?

Reply:
each packet is of 512 bytes, so allocating maximum of 512 bytes any given time. Should have checked for malloc failure.